### DISCUSSION TITLE
Check if record already exists, if not create it, and create a m-n relationship

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9127] [Q&A] [answered]
I have the following mutation:

```ts
    const moviesAlreadyExists = await prisma.movie.findUnique({
      where: {tmdb_id: args.tmdb_id}});

    if (!moviesAlreadyExists) {
      await prisma.movie.create({
        data: {
          original_title: args.original_title,
          tmdb_id: args.tmdb_id,
          poster_path: args.poster_path,
          users: {
            connect: [{id: req.userId}],
          },
        },
      });
    }

    if (moviesAlreadyExists) {
      await prisma.movie.update({
        where: {tmdb_id: args.tmdb_id},
        data: {
          users: {
            connect: [{id: req.userId}],
          },
        },
      });
    }
```
First I check if a movie with the specific `tmdb_id` already exists. If not I create it and connect the current user to it. If it does exist I update the movie to connect it to the user.

I think I can use the [connectOrCreate](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-or-create-a-record) api for this, but I can't figure out the correct code because the example:

```ts
const createPost = await prisma.post.create({
  data: {
    title: 'How to make croissants',
    author: {
      connectOrCreate: {
        where: {
          email: 'viola@prisma.io',
        },
        create: {
          email: 'viola@prisma.io',
          name: 'Viola',
        },
      },
    },
  },
  include: {
    author: true,
  },
})
```
Will always create a new post, and it will connect it to a user if it exists, otherwise create a user. In my case I don't always want to create a movie and never a user.

**Comment by ryands17** [comment, marked as answer]
@alucardu 👋 

What you need is an [upsert](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert).

**Reply by alucardu** [reply]
```
    await prisma.movie.upsert({
      where: {
        tmdb_id: args.tmdb_id,
      },
      update: {
        users: {
          connect: [{id: req.userId}],
        },
      },
      create: {
        original_title: args.original_title,
        tmdb_id: args.tmdb_id,
        poster_path: args.poster_path,
        users: {
          connect: [{id: req.userId}],
        },
      },
    });
```

Fixed it for me, thanks @ryands17 for the quick help. Prisma is really really great, as a front-end dev using Prisma and Apollo I can quickly build a API that's manageable!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
